gio/src/list_model.rs

```diff
@@ -19,15 +30,86 @@ impl<T: IsA<ListModel>> ListModelExtManual for T {
         }
         res
     }
+    fn iter<LT: IsA<glib::Object>>(&self) -> Result<ListModelIter<LT>, &Self>
+    where
+        Self: Sized + Clone,
+    {
+        if self.item_type() != LT::static_type() {
+            return Err(self);
+        }
+        let rc = Rc::new(Cell::new(false));
+
+        let rcc = rc.clone();
+        let signal_id = Cell::new(Some(self.connect_items_changed(move |_, _, _, _| {
+            rcc.replace(true);
+        })));
+
+        Ok(ListModelIter {
+            ty: Default::default(),
+            i: 0,
+            model: self.clone().upcast(),
```

Comment:
Why not just store a reference in here instead of cloning?

Reply:
because `upcast` consumes `self`:
`pub fn upcast<T>(self)`

Reply:
You could do `model: self.upcast_ref(),` and just store a reference. That also makes it less strange as iterators usually keep a normal reference to their underlying data